### ISSUE TITLE
Compilation problems

### ISSUE THREAD

**Issue by henrykmichalewski** [dreal/dreal#93]
On Ubuntu 14.04 I ran into compilation issues with Gflags package. Namely, it seems that automatic update of the Gflags repository fails. I installed separetly Gflags from sources (without problems), but it did not help in the sense that the Makefile still tries to install Gflags on its own and fails. 

```
[ 13%] Performing download step (git clone) for 'GFLAGS'
Cloning into 'GFLAGS'...
remote: Counting objects: 1696, done.
remote: Compressing objects: 100% (130/130), done.
remote: Total 1696 (delta 65), reused 0 (delta 0), pack-reused 1566
Receiving objects: 100% (1696/1696), 1.24 MiB | 749.00 KiB/s, done.
Resolving deltas: 100% (971/971), done.
Checking connectivity... done.
Note: checking out 'origin/master'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at c354a1b... fix(gflags_completions.cc): remove field-initializers warnings
Submodule 'doc' (git@github.com:gflags/gflags.git) registered for path 'doc'
Cloning into 'doc'...
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Clone of 'git@github.com:gflags/gflags.git' into submodule path 'doc' failed
CMake Error at /home/henryk/dreal/build/release/external/tmp/GFLAGS-gitclone.cmake:67 (message):
  Failed to update submodules in:
  '/home/henryk/dreal/build/release/external/src/GFLAGS'


make[2]: *** [external/src/GFLAGS-stamp/GFLAGS-download] Error 1
make[1]: *** [CMakeFiles/GFLAGS.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by soonhokong** [comment]
Could you do the following and let me know if it fixes the problem?

 - move to the build directory
 - `rm -rf external lib include`
 - `make` (or `ninja`) to rebuild

I recently made a distructive update on dreal-deps/gflags repository and I think the old one interfered with the new one.

**Comment by soonhokong** [comment]
@henrykmichalewski, I just push a commit 02d385f28193a8513559143326e3d85a8a9fcfbb and I think it fixes the problem. Could you do 'make clean' and rebuild dReal?

**Comment by henrykmichalewski** [comment]
I did the following steps: removed completely the old version of dreal and cloned again the github repository. I followed the ubuntu installation instructions from the github page

```
git clone https://github.com/dreal/dreal.git dreal
cd dreal
mkdir -p build/release
cd build/release
cmake -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_CXX_COMPILER=g++-4.8 -DCMAKE_C_COMPILER=gcc-4.8 ../../src
make
```

Finally the whole process is still stopping at GFLAGS

```
remote: Counting objects: 1696, done.
remote: Compressing objects: 100% (130/130), done.
remote: Total 1696 (delta 65), reused 0 (delta 0), pack-reused 1566
Receiving objects: 100% (1696/1696), 1.24 MiB | 1.03 MiB/s, done.
Resolving deltas: 100% (971/971), done.
Checking connectivity... done.
Note: checking out 'origin/master'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at c354a1b... fix(gflags_completions.cc): remove field-initializers warnings
Submodule 'doc' (git@github.com:gflags/gflags.git) registered for path 'doc'
Cloning into 'doc'...
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Clone of 'git@github.com:gflags/gflags.git' into submodule path 'doc' failed
CMake Error at /home/henryk/dreal/build/release/external/tmp/GFLAGS-gitclone.cmake:67 (message):
  Failed to update submodules in:
  '/home/henryk/dreal/build/release/external/src/GFLAGS'


make[2]: *** [external/src/GFLAGS-stamp/GFLAGS-download] Error 1
make[1]: *** [CMakeFiles/GFLAGS.dir/all] Error 2
make: *** [all] Error 2
```
Admittedly, if I simply try to use the script build-dreal.sh in the main directory, then it fails on something else (I did not investigate it in details, there are actually many errors).

**Comment by soonhokong** [comment]
I'm on it...

**Comment by soonhokong** [comment]
One of the problems is https://github.com/gflags/gflags/issues/112. I made a patch on our repository https://github.com/dreal-deps/gflags/commit/28d651ccab0a7b78448585fae89f26ea7846065c

I'm still testing and solving other issues. I'll let you know when everything is working.

**Comment by soonhokong** [comment]
@henrykmichalewski, could you check out the latest version and test? I've also done it on a fresh Ubuntu-14.04 VM and it worked well there. I also fixed some dependency issues by 84482116c836e428be60d14793c9b026a222b673

**Comment by henrykmichalewski** [comment]
dReal compiles and works fine.Thank you!
